### PR TITLE
Expose stream-ordering in subword tokenizer API

### PR DESCRIPTION
**Pull request by shrshi**
## Description
Add stream parameter to public APIs:
```
nvtext::subword_tokenize
nvtext::load_vocabulary_file
```
Added stream gtest.

Reference: #13744 

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/tests/CMakeLists.txt [hunk 1]
```diff
@@ -741,6 +741,7 @@ ConfigureTest(
   streams/text/ngrams_test.cpp
   streams/text/replace_test.cpp
   streams/text/stemmer_test.cpp
+  streams/text/subword_tokenize_test.cpp
```
Comment (mhaseeb123) [inline cpp/tests/CMakeLists.txt:744]: Does this file exist? I think you might have forgotten to add it to the PR?

Comment (shrshi) [reply cpp/tests/CMakeLists.txt:744]: omg, I was wondering how it builds on my machine lol
Added now!

File: cpp/tests/streams/text/subword_tokenize_test.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+// The vocab only includes the following words:
+//  'this', 'is', 'a', 'test', 'tést'
+// The period '.' character also has a token id.
+void create_hashed_vocab(std::string const& hash_file)
+{
+  std::vector<std::pair<int, int>> coefficients(23, {65559, 0});
+  std::ofstream outfile(hash_file, std::ofstream::out);
+  outfile << "1\n0\n" << coefficients.size() << "\n";
+  for (auto c : coefficients)
+    outfile << c.first << " " << c.second << "\n";
```
Comment (mhaseeb123) [inline cpp/tests/streams/text/subword_tokenize_test.cpp:46]:
```suggestion
  for (auto c : coefficients) {
    outfile << c.first << " " << c.second << "\n";
  }
```

File: cpp/tests/streams/text/subword_tokenize_test.cpp [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  std::vector<uint64_t> hash_table(23, 0);
+  outfile << hash_table.size() << "\n";
+  hash_table[0]  = 3015668L;              // based on values
+  hash_table[1]  = 6205475701751155871L;  // from the
+  hash_table[5]  = 6358029;               // bert_hash_table.txt
+  hash_table[16] = 451412625363L;         // file for the test
+  hash_table[20] = 6206321707968235495L;  // words above
+  for (auto h : hash_table)
+    outfile << h << "\n";
```
Comment (mhaseeb123) [inline cpp/tests/streams/text/subword_tokenize_test.cpp:55]:
```suggestion
  for (auto h : hash_table) {
    outfile << h << "\n";
  }
```

File: cpp/tests/streams/text/subword_tokenize_test.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  uint32_t max_sequence_length = 16;
+  uint32_t stride              = 16;
+
+  auto result = nvtext::subword_tokenize(cudf::strings_column_view{strings},
+                                         *vocab,
+                                         max_sequence_length,
+                                         stride,
+                                         true,   // do_lower_case
+                                         false,  // do_truncate
+                                         cudf::test::get_default_stream());
```
Comment (mhaseeb123) [inline cpp/tests/streams/text/subword_tokenize_test.cpp:77]: Shouldn't we be asserting on or checking something in the test?

File: cpp/tests/streams/text/subword_tokenize_test.cpp [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+struct TextSubwordTest : public cudf::test::BaseFixture {};
+
+// Create a fake hashed vocab text file for the tests in this source file.
+// The vocab only includes the following words:
+//  'this', 'is', 'a', 'test', 'tést'
+// The period '.' character also has a token id.
+void create_hashed_vocab(std::string const& hash_file)
+{
+  std::vector<std::pair<int, int>> coefficients(23, {65559, 0});
```
Comment (mhaseeb123) [inline cpp/tests/streams/text/subword_tokenize_test.cpp:42]: Minor nit: The magic number `23` is seen a few times as we are writing it as `coefficients.size()` and `hash_table.size()` as well, so can we createa and use a `constexpr` variable instead?

File: cpp/tests/streams/text/subword_tokenize_test.cpp [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    outfile << h << "\n";
+  outfile << "100\n101\n102\n\n";
+}
+
+TEST(TextSubwordTest, Tokenize)
+{
+  uint32_t nrows = 100;
+  std::vector<char const*> h_strings(nrows, "This is a test. A test this is.");
+  cudf::test::strings_column_wrapper strings(h_strings.begin(), h_strings.end());
+  std::string hash_file = temp_env->get_temp_filepath("hashed_vocab.txt");
```
Comment (mhaseeb123) [inline cpp/tests/streams/text/subword_tokenize_test.cpp:64]: Some missing consts
```suggestion
  uint32_t const nrows = 100;
  std::vector<char const*> h_strings(nrows, "This is a test. A test this is.");
  cudf::test::strings_column_wrapper strings(h_strings.cbegin(), h_strings.cend());
  std::string const hash_file = temp_env->get_temp_filepath("hashed_vocab.txt");
```

**Review by mhaseeb123 — CHANGES_REQUESTED** [review]
Some minor nits

File: cpp/tests/streams/text/subword_tokenize_test.cpp (hunk 4 above)
Comment (davidwendt) [reply cpp/tests/streams/text/subword_tokenize_test.cpp:77]: No. The stream tests automatically track the test stream is passed through and used correctly.
These tests are not meant to check for correctness -- only the stream is passed correctly.

Comment (shrshi) [reply cpp/tests/streams/text/subword_tokenize_test.cpp:77]: Thank you, @davidwendt!
I simplified the test for this reason - we just need to ensure that `subword_tokenize` does not throw an unexpected stream exception. It is sufficient to invoke the function with the test stream without checking the functional correctness of it.

**Review by ttnghia — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]

**Review by mhaseeb123 — APPROVED** [review]

**Pull request merged**